test/src/loaders/workspace_uploader/workspace_uploader_test.py

```diff
@@ -368,14 +529,152 @@ def test_generator(setup_and_teardown):
+                     "assembly_info": ASSEMBLY_OBJ_INFOS[0],
+                     "genome_info": GENOME_OBJ_INFOS[0]},
+                    {"genome_ref": genome_refs[1],
+                     "assembly_ref": assembly_refs[1],
+                     "assembly_path": container_dir / ASSEMBLY_NAMES[1],
+                     "assembly_info": ASSEMBLY_OBJ_INFOS[1],
+                     "genome_info": GENOME_OBJ_INFOS[1]}]
+    }
+    gfu.genbanks_to_genomes.return_value = genbanks_to_genomes_results
+
+    with patch.object(workspace_uploader, '_JOB_DIR_IN_ASSEMBLYUTIL_CONTAINER', new=container_dir):
+        uploaded_count = workspace_uploader._upload_objects_in_parallel(
+            ws=ws,
+            upload_env_key="CI",
+            workspace_id=72231,
+            load_id="214",
+            collections_source_dir=collection_source_dir,
+            wait_to_upload_objs=wait_to_upload_genomes,
+            batch_size=2,
+            source_data_dir=sourcedata_dir,
+            asu_client=Mock(),
+            gfu_client=gfu,
+            job_data_dir=container_dir,
+            upload_assembly_only=False,
+        )
+
+    assert uploaded_count == 2
+
+    # assert that no interactions occurred with ws
+    ws.get_object_info3.assert_not_called()
+
+    # assert that asu was called correctly
+    gfu.genbanks_to_genomes.assert_called_once_with(
+        {
+            "workspace_id": 72231,
+            "inputs": [
+                {
+                    "file": {"path": f"{container_dir}/GCF_000979855.1_gtlEnvA5udCFS_genomic.gbff.gz"},
+                    "genome_name": "GCF_000979855.1_gtlEnvA5udCFS_genomic.gbff.gz",
+                    "metadata": {"load_id": "214"},
+                },
+                {
+                    "file": {"path": f"{container_dir}/GCF_000979175.1_gtlEnvA5udCFS_genomic.gbff.gz"},
+                    "genome_name": "GCF_000979175.1_gtlEnvA5udCFS_genomic.gbff.gz",
+                    "metadata": {"load_id": "214"},
+                }
+            ]
+        }
+    )
+
+    assembly_dirs = [obj_info_to_upa(info, underscore_sep=True) for info in ASSEMBLY_OBJ_INFOS]
+    genome_dirs = [obj_info_to_upa(info, underscore_sep=True) for info in GENOME_OBJ_INFOS]
+
+    # check softlink for post_process
+    assert (os.readlink(os.path.join(collection_source_dir, assembly_dirs[0]))
+            == os.path.join(sourcedata_dir, assembly_dirs[0]))
+
+    assert (os.readlink(os.path.join(collection_source_dir, assembly_dirs[0]))
+            == os.path.join(sourcedata_dir, assembly_dirs[0]))
+
+    # check hardlink for post_process
+    assert os.path.samefile(
+        genome_collection_source_dirs[0] / ASSEMBLY_NAMES[0],
+        os.path.join(sourcedata_dir, assembly_dirs[0], f"{assembly_dirs[0]}.fna.gz")
+    )
+
+    assert os.path.samefile(
+        genome_collection_source_dirs[1] / ASSEMBLY_NAMES[1],
+        os.path.join(sourcedata_dir, assembly_dirs[1], f"{assembly_dirs[1]}.fna.gz")
+    )
```

Comment:
These are checking collectoinssource/NONE vs sourcedir right, which are softlinked dirs? It should be collectionssource/ENV?

Reply:
No. collectoinsource/NONE and source/UPA are not softlinked dirs. It's testing:
https://github.com/kbase/collections/blob/311aef80e6a14bdd32540b30b39db03109499c02/src/loaders/workspace_uploader/workspace_uploader.py#L624-L625

I think the test is correct.

Reply:
genome_collection_source_dirs isn't the workspace object directory though, unless I'm confused

Reply:
Oh I see, assembly_dirs[1] is a underscore UPA. Can that have a clearer name?

Reply:
`os.path.join(sourcedata_dir, assembly_dirs[1])` is the workspace object directory. `genome_collection_source_dirs` are collectoinsource dir.

Reply:
👍

Reply:
:+1: